boilermaker_desktop/src/ui/new_project.rs

```diff
@@ -0,0 +1,217 @@
+use std::collections::HashMap;
+use std::path::PathBuf;
+
+use boilermaker_desktop::APP_STATE;
+use color_eyre::Result;
+use dioxus::prelude::*;
+
+use crate::Route;
+use boilermaker_core::commands::new as Command;
+use boilermaker_core::db::TemplateResult;
+use boilermaker_core::template as tpl;
+use boilermaker_core::template::static_analysis::find_variables_in_path;
+use boilermaker_views::{BTN_GREEN_STYLE, INPUT_STYLE, LABEL_STYLE, LINK_STYLE};
+
+// TODO: Share relevant enums across UI elements. This is also used in TemplateAdd.
+enum ResultMessage {
+    None,
+    Error(String),
+    Success(String),
+}
+
+struct FormState {
+    dir: String,
+    overwrite: bool,
+    rename: String,
+    variables: Vec<String>,
+    variable_values: HashMap<String, String>,
```

Comment:
This should really be a single `variables: HashMap<String, String>` attribute. I'll fix that before merging. ;)